options/posix/generic/unistd-stubs.cpp

```diff
@@ -194,14 +201,29 @@ int faccessat(int dirfd, const char *pathname, int mode, int flags) {
 }
 
 int fchown(int fd, uid_t uid, gid_t gid) {
-	mlibc::infoLogger() << "\e[31mmlibc: fchown() is not implemented correctly\e[39m"
-			<< frg::endlog;
+	if(!mlibc::sys_fchownat) {
+		MLIBC_MISSING_SYSDEP();
+		errno = ENOSYS;
+		return -1;
+	}
+	if(int e = mlibc::sys_fchownat(fd, nullptr, uid, gid, 0, true); e) {
+		errno = e;
+		return -1;
+	}
 	return 0;
 }
 
 int fchownat(int fd, const char *path, uid_t uid, gid_t gid, int flags) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+	if(!mlibc::sys_fchownat) {
+		MLIBC_MISSING_SYSDEP();
+		errno = ENOSYS;
+		return -1;
+	}
+	if(int e = mlibc::sys_fchownat(fd, path, uid, gid, flags, true); e) {
```

Comment:
Here we need to pass `AT_EMPTY_PATH`, don't we?

Reply:
I don't think we do, if the application wants that behavior, it should add `AT_EMPTY_PATH` to flags itself, right?

Reply:
Oh, I attached the comment in the wrong function. I meant `fchown`. fchown is basically `fchownat` with AT_EMPTY_PATH, isn't it?

Reply:
Hmm, that seems possible yes, I'll change that.